Support for generating advisories with logout suggestions

Conan-Kudo opened the pull request:
Updateinfo metadata for RPM repositories can include information on whether
the user needs to log out and log back in. However, Bodhi only partially
supported this as a UI flag that wasn't set in the metadata. Now we
will propagate this into the metadata for clients to use.

Fixes #4213

Conan-Kudo commented:
cc: @rdieter

Conan-Kudo committed:
Support for generating advisories with logout suggestions

Updateinfo metadata for RPM repositories can include information on whether
the user needs to log out and log back in. However, Bodhi only partially
supported this as a UI flag that wasn't set in the metadata. Now we
will propagate this into the metadata for clients to use.

Signed-off-by: Neal Gompa <ngompa13@gmail.com>

mattiaverga commented on bodhi/server/metadata.py:
```diff
@@ -258,8 +258,8 @@ def add_update(self, update):
 
-                # TODO: how do we handle UpdateSuggestion.logout, etc?
                 pkg.reboot_suggested = update.suggest is UpdateSuggestion.reboot
+                pkg.relogin_suggested = update.suggest is UpdateSuggestion.logout
```
Since LGTM complains about the usage of `is` instead of `==`, can you change those two lines?

Otherwise... LGTM!

Conan-Kudo replied:
Done in a followup commit!

Conan-Kudo committed:
server/metadata: Test equivalance for updateinfo flags

We should test for equivalence properly for updateinfo flags rather
than relying on identity tests, which may not work as intended.

Signed-off-by: Neal Gompa <ngompa13@gmail.com>

mattiaverga commented:
Thanks, pip-* tests and f33-integration failures are not related to these changes.

Conan-Kudo commented:
@mattiaverga Any idea when this will be merged and released?

mattiaverga commented:
> @mattiaverga Any idea when this will be merged and released?

No idea, I don't know who's the primary point of contact for Bodhi nowadays.

travier commented:
Ping here. Could we get this reviewed / merged? Thanks!

lgriffin commented:
@Zlopez let's add this to our pile

Zlopez commented:
@Conan-Kudo Could you rebase this to latest develop? The CI should be in much better state now.

AdamSaleh commented:
Cherry-picked and merged in https://github.com/fedora-infra/bodhi/pull/4254
